In [6]:
import numpy as np
import pandas as pd
%matplotlib inline 

housing = pd.read_csv('Ames_Housing_Price_Data.csv', index_col=0)


In [8]:
housing.sample(10)

PID  GrLivArea  SalePrice  MSSubClass MSZoning  LotFrontage  \
565  527302090       1664     213000          20       RL         93.0   
713  911370540       1646     275000         120       RM         44.0   
431  527302210       1782     220000          20       RL         85.0   
312  905107070       1400     149500          60       RL         74.0   
4    534125120       1152     143450          20       RL         78.0   
459  534225110       1728     145000          90       RL         72.0   
908  535353040       1031     145500          20       RL         60.0   
381  533254100       1855     187000          80       RL         80.0   
212  909428190       1640     224000          20       RL          NaN   
178  903206170       1635     210000          50       RL         70.0   

     LotArea Street Alley LotShape  ... ScreenPorch PoolArea PoolQC  Fence  \
565    12299   Pave   NaN      Reg  ...           0        0    NaN    NaN   
713     3842   Pave   NaN      IR1  ...         155        0    NaN    NaN   
431    14450   Pave   NaN      Reg  ...           0        0    NaN    NaN   
312     7844   Pave   NaN      Reg  ...           0        0    NaN    NaN   
4      11700   Pave   NaN      Reg  ...           0        0    NaN    NaN   
459    11072   Pave   NaN      IR1  ...           0        0    NaN    NaN   
908     7626   Pave   NaN      Reg  ...           0        0    NaN  MnPrv   
381     9600   Pave   NaN      Reg  ...           0        0    NaN    NaN   
212    14778   Pave   NaN      IR1  ...           0        0    NaN    NaN   
178     8737   Pave   NaN      IR1  ...           0        0    NaN    NaN   

    MiscFeature MiscVal MoSold YrSold SaleType  SaleCondition  
565         NaN       0      9   2007      WD          Normal  
713         NaN       0      1   2008      WD          Normal  
431         NaN       0      3   2008      WD          Normal  
312         NaN       0      3   2006      WD          Normal  
4           NaN       0      9   2007      CWD         Normal  
459         NaN       0     11   2008      WD          Normal  
908         NaN       0      3   2009      WD          Normal  
381         NaN       0      8   2007      WD          Normal  
212         NaN       0     11   2008      WD          Normal  
178         NaN       0      5   2007      WD          Normal  

[10 rows x 81 columns]

In [10]:
housing.columns

Index(['PID', 'GrLivArea', 'SalePrice', 'MSSubClass', 'MSZoning',
       'LotFrontage', 'LotArea', 'Street', 'Alley', 'LotShape', 'LandContour',
       'Utilities', 'LotConfig', 'LandSlope', 'Neighborhood', 'Condition1',
       'Condition2', 'BldgType', 'HouseStyle', 'OverallQual', 'OverallCond',
       'YearBuilt', 'YearRemodAdd', 'RoofStyle', 'RoofMatl', 'Exterior1st',
       'Exterior2nd', 'MasVnrType', 'MasVnrArea', 'ExterQual', 'ExterCond',
       'Foundation', 'BsmtQual', 'BsmtCond', 'BsmtExposure', 'BsmtFinType1',
       'BsmtFinSF1', 'BsmtFinType2', 'BsmtFinSF2', 'BsmtUnfSF', 'TotalBsmtSF',
       'Heating', 'HeatingQC', 'CentralAir', 'Electrical', '1stFlrSF',
       '2ndFlrSF', 'LowQualFinSF', 'BsmtFullBath', 'BsmtHalfBath', 'FullBath',
       'HalfBath', 'BedroomAbvGr', 'KitchenAbvGr', 'KitchenQual',
       'TotRmsAbvGrd', 'Functional', 'Fireplaces', 'FireplaceQu', 'GarageType',
       'GarageYrBlt', 'GarageFinish', 'GarageCars', 'GarageArea', 'GarageQual',
       'GarageCond

In [26]:
housing.groupby('Neighborhood').agg({'PID':'count'}).sort_values(by= 'PID', ascending=False).reset_index()

Neighborhood  PID
0         NAmes  410
1       CollgCr  236
2       OldTown  213
3       Edwards  165
4       Somerst  143
5       Gilbert  143
6        Sawyer  139
7        NWAmes  123
8       NridgHt  121
9       SawyerW  113
10      Mitchel  104
11      BrkSide  103
12      Crawfor   93
13       IDOTRR   76
14      NoRidge   67
15       Timber   54
16      StoneBr   43
17        SWISU   42
18      ClearCr   40
19      MeadowV   34
20       BrDale   29
21      Blmngtn   23
22      Veenker   23
23      NPkVill   22
24      Blueste   10
25       Greens    8
26      GrnHill    2
27      Landmrk    1

In [30]:
housing.shape

(2580, 81)

In [36]:
np.mean(housing.loc[~housing.PoolQC.isnull()].SalePrice)

255944.44444444444

Average _Price_ pool = $255,944   , 9 houses


Average_Price _without_pool = $177,786.98  , 2571 Houses

In [39]:
np.mean(housing.loc[housing.PoolQC.isnull()].SalePrice)

177786.98094126798

In [40]:
housing.GarageCond.unique()

array(['TA', 'Po', nan, 'Fa', 'Gd', 'Ex'], dtype=object)

In [41]:
np.mean(housing.loc[housing.GarageCond.isnull()].SalePrice)

106814.6046511628

In [47]:
housing.loc[housing.GarageCond.isnull()].count()

PID              129
GrLivArea        129
SalePrice        129
MSSubClass       129
MSZoning         129
                ... 
MiscVal          129
MoSold           129
YrSold           129
SaleType         129
SaleCondition    129
Length: 81, dtype: int64

In [48]:
np.mean(housing.loc[housing.GarageCond.isnull()].SalePrice)

106814.6046511628

In [49]:
housing.loc[~housing.GarageCond.isnull()].count()

PID              2451
GrLivArea        2451
SalePrice        2451
MSSubClass       2451
MSZoning         2451
                 ... 
MiscVal          2451
MoSold           2451
YrSold           2451
SaleType         2451
SaleCondition    2451
Length: 81, dtype: int64

In [52]:
np.mean(housing.loc[~housing.GarageCond.isnull()].SalePrice)

181809.3610771114

In [53]:
housing.BsmtCond.unique()

array(['TA', nan, 'Fa', 'Gd', 'Ex', 'Po'], dtype=object)

In [55]:
housing.loc[housing.BsmtCond.isnull()].count()

PID              69
GrLivArea        69
SalePrice        69
MSSubClass       69
MSZoning         69
                 ..
MiscVal          69
MoSold           69
YrSold           69
SaleType         69
SaleCondition    69
Length: 81, dtype: int64